test/controllers/authentications_controller_test.rb

```diff
@@ -1,30 +1,38 @@
+# frozen_string_literal: true
+
 require 'test_helper'
 
 class AuthenticationsControllerTest < ActionDispatch::IntegrationTest
   test 'when logging in fails because the email is missing' do
-    post login_url
+    assert_no_difference ['Session.count'] do
```

Comment:
I wrapped all of these unsuccessful login tests with a no-difference assertion on the `Session` table's record count